### ISSUE TITLE
Enable other tunable parameters for `create_optimizer` in `optimization.py`

### ISSUE THREAD

**Issue by 01-vyom** [tensorflow/models#10143]
# Prerequisites

- [X] I checked to make sure that this feature has not been requested already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/blob/7ecbac3c3bb727cb9f6170476a6957465cffdee1/official/nlp/optimization.py#L68-L107

## 2. Describe the feature you request

Similar to `beta_1` as a tunable parameter for creating optimizers, we can add `beta_2`, `epsilon`, `weight_decay_rate`, and `exclude_from_weight_decay` as tunable parameters by passing them as argument from `create_optimizer`.

## 3. Additional context

I was recently trying to finetune a [Huggingface](https://github.com/huggingface/transformers) Roberta model and while doing so, I wanted to add a scheduler as well as AdamW with custom parameters, and thus I came across these methods. 

## 4. Are you willing to contribute it? (Yes or No)

Yes

**Comment by saberkun** [comment]
Hi, we are using the configurable optimization now: https://github.com/tensorflow/models/tree/master/official/modeling/optimization
All params should be configurable now. It is not used in the legacy bert/ example.

**Comment by 01-vyom** [comment]
Nice!! I was following this guide for [fine-tuning TF BERT](https://www.tensorflow.org/official_models/fine_tuning_bert#set_up_the_optimizer), and the `official nlp optimizer` was used to set up the optimizer. So, if this configurable optimization is ready, where can I find some examples for that? or can we change the above-mentioned fine-tuning example by using this configurable optimization? Let me know what you think about this.

**Comment by saberkun** [comment]
Here is the unit test: https://github.com/tensorflow/models/blob/master/official/modeling/optimization/optimizer_factory_test.py
Here is the use case: 
https://github.com/tensorflow/models/blob/master/official/core/base_task.py#L67

The bert tutorial was limited to the particular use case as a demo.
Yes, we will consider updating the colab which is a bit old already.

**Comment by 01-vyom** [comment]
Thank you for the resources!! Looking forward to the update in the Colab example. Let me know if I can contribute in any way.

This issue can be closed.
